hack/build/build-go.sh

```diff
@@ -62,6 +62,11 @@ elif [ "${go_opt}" == "build" ]; then
             ln -sf ${outFile} ${outLink}
         )
     done
+
+    # This container also needs the csv-generator file, so it needs to be
+    # copied into the operator's directory for docker build to source
+    # csv-generator correctly
+    cp ${OUT_DIR}/tools/${CSV_GENERATOR}/${CSV_GENERATOR} ${OUT_DIR}/cmd/${OPERATOR}/${CSV_GENERATOR}
```

Comment:
I wonder if this would be better handled elsewhere?  I don't think it's a big deal because we typically build everything but technically `${OUT_DIR}/tools/${CSV_GENERATOR}/${CSV_GENERATOR}` is not always guaranteed to exist.

Reply:
yeah, this is an unfortunate hack. I couldn't figure out another place to put it... I could add a check to only copy it when OUT_DIR/tools/CSV_GENERATOR/CSV_GENERATOR exists to make it safer though. Would that make sense?

Reply:
It is possible add this copy to ``` ./hack/build/build-copy-artifacts.sh``` as it is done for cloner, or as another option (I prefer this one ) to add a dedicated script ```hack/build/build-cdi-operator.sh ``` that should be added to build target in Makefile and will copy all the required artifacts to proper location

Reply:
General question - how HCO will execute csv-generator binary? will it exec -ti to operator container ?

Reply:
This binary will be executed during the HCO's build process when the unified CSV is generated. so, something like `docker run -it -rm --entrypoint=/usr/bin/csv-generator <imagename> <args>`

The issue is the HCO doesn't currently have the ability to accurately populate what cdi operators's Deployment, RBAC, CRDs need to be in the unified CSV file because these values have the potential to change between every CDI build. To fix this, we're using the build artifact  (cdi-operator container image in this case) to be the source of truth.

Basically, the HCO knows what cdi-operator container image it is integrating with, so it can then use that container image to "discover" the Deployement, RBAC rules, CRDs associated with the cdi-operator by executing the csv-generator binary/script.